src/main.ts

```diff
@@ -25,7 +25,7 @@ const mergeObject = (a: Param, b: Param): Response => {
       return { ...acc, [field]: mergeArray(aValue, bValue) }
     }
 
-    if (bValue !== undefined && !isObject(bValue)) {
+    if (bValue !== undefined && !isObject(bValue)) { // TODO: Merge only when bValue is a non-complex type. Currently, bValue can be anything but an object, and it will be stored in acc[field]
```

Comment:
Good catch! I just updated this validation, please let me know your thoughts. Thank you!

Reply:
New version looks great. Keep the good work.